### PR TITLE
Add printcolumns support on generated resources

### LINKED ISSUE #1665: Add kubebuilder:printcolumn directives to generated resources
**Describe the current behavior**
At the moment generated resources don't indicate their provisioning state in tabular kubectl output, so you need to keep checking the yaml output to find out whether they're ready.

**Describe the improvement**
Adding `kubebuilder:printcolumn` comments to the generated code would make them much nicer to use:
```
// +kubebuilder:printcolumn:name="ProvisioningState",type="string",JSONPath=".status.provisioningState"
```
(Although the name `ProvisioningState`'s probably a bit long.)

At the moment there's not really an equivalent of the `message` field on the ASOv1 resources - once the condition work spec'd in #1614 is done it would make sense to use the status and message fields from the ready condition (presumably these will supplant the `status.provisioningState`).

### PR DESCRIPTION
**Pull request by matthchr**
Closes #1665

**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

**Comment by matthchr** [comment]
I need to actually test this in kubectl before I merge - my `kind` was broken on Friday.

**Review by babbageclunk — APPROVED** [review]
Yay, these will be great!

File: hack/generator/pkg/astmodel/resource_type.go [hunk 1]
```diff
@@ -286,6 +288,14 @@ func (resource *ResourceType) Equals(other Type) bool {
 		}
 	}
 
+	// Check same annotations present
```
Comment (theunrepentantgeek) [inline hack/generator/pkg/astmodel/resource_type.go:291]:
```suggestion
	// Check same annotations present in the same order
```

**Review by theunrepentantgeek — APPROVED** [review]
Looks good.

gif

**Review by Porges — APPROVED** [review]

File: hack/generator/pkg/astmodel/resource_type.go (hunk 1 above)
Comment (matthchr) [reply hack/generator/pkg/astmodel/resource_type.go:291]: Added

**Comment by matthchr** [comment]
Confirmed this works:
```
~/work/github/azure-service-operator (feature/printcolumns) $ k get storageaccounts
NAME                READY   REASON        MESSAGE
samplekubestorage   False   Reconciling   The resource is in the process of being reconciled by the operator
```

```
~/work/github/azure-service-operator (feature/printcolumns) $ k get storageaccounts
NAME                READY   REASON      MESSAGE
samplekubestorage   True    Succeeded   
```

**Pull request merged**